api/python/provisioner/commands/check.py

```diff
@@ -609,6 +616,177 @@ def _cluster_status(*, args: str) -> CheckEntry:
+        """
+
+        res: List[CheckEntry] = list()
+        driver_name = [cfg.NETWORK_DRIVER]
+
+        if cortx_py_utils_import_error:
+            err_res : CheckEntry = CheckEntry(cfg.Checks.NETWORK_DRIVERS.value)
+            err_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                             comment="Package cortx-py-utils not installed")
+            res.append(err_res)
+            return res
+
+        check_value : CheckEntry = CheckEntry(cfg.Checks.NETWORK_DRIVERS.value)
+        try:
+            nodes = Check._get_pillar_data("cluster/node_list")
+            check_res : CheckEntry = CheckEntry(
+                                     cfg.Checks.NETWORK_DRIVERS.value)
+            try:
+                driver_args = driver_name + nodes
+                NetworkV().validate('drivers', driver_args)
+                check_res.set_passed(checked_target=cfg.ALL_MINIONS,
+                                     comment="Network Driver "
+                                     f"Validated: {driver_args}")
+
+            except Exception as exc:
+                check_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                                   comment=str(exc))
+            res.append(check_res)
+
+        except Exception as exc:
+            check_value.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            res.append(check_value)
+
+        return res
+
+    @staticmethod
+    def _network_hca(*, args: str) -> List[CheckEntry]:
+        """Validates the Network HCA
+        network_hca <provider_name> <nodes>
+        """
+
+        res: List[CheckEntry] = list()
+        provider = [cfg.HCA_PROVIDER]
+
+        if cortx_py_utils_import_error:
+            err_res : CheckEntry = CheckEntry(cfg.Checks.NETWORK_HCA.value)
+            err_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                             comment="Package cortx-py-utils not installed")
+            res.append(err_res)
+            return res
+
+        check_value : CheckEntry = CheckEntry(cfg.Checks.NETWORK_HCA.value)
+        try:
+            nodes = Check._get_pillar_data("cluster/node_list")
+            check_res : CheckEntry = CheckEntry(cfg.Checks.NETWORK_HCA.value)
+            try:
+                hca_args = provider + nodes
+                NetworkV().validate('hca', hca_args)
+                check_res.set_passed(checked_target=cfg.ALL_MINIONS,
+                                     comment="HCA Presence and "
+                                     f"Ports Validated: {hca_args}")
+
+            except Exception as exc:
+                check_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                                   comment=str(exc))
+            res.append(check_res)
+
+        except Exception as exc:
+            check_value.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            res.append(check_value)
+
+        return res
```

Comment:
This method also can be optimized as I mentioned for ` _network_drivers` method.

Reply:
Comment addressed.

Reply:
We can avoid the nested `try-except` as following. Also you forgot to move 
```
                check_res.set_passed(checked_target=cfg.ALL_MINIONS,
                                     comment="HCA Presence and "
                                     f"Ports Validated: {hca_args}")
```
into `else`-part of `try-except-else` block.
So as result method can be simplified as
```python
@staticmethod
def _network_hca(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
    """
    Check if Network HCA are proper
    :param args: network_hca check specific parameters and arguments
    :return:
    """
    check_value: CheckEntry = CheckEntry(cfg.Checks.NETWORK_HCA.value)

    if cortx_py_utils_import_error:
        check_value.set_fail(checked_target=cfg.ALL_MINIONS,
                         comment="Package cortx-py-utils not installed")
        return check_value

    try:
        nodes = Check._get_pillar_data("cluster/node_list")
    except Exception as exc:
        check_value.set_fail(checked_target=cfg.ALL_MINIONS,
                             comment=str(exc))
        return check_value

    res: List[CheckEntry] = list()
    for provider in cfg.HCA_PROVIDER:
        check_res: CheckEntry = CheckEntry(cfg.Checks.NETWORK_HCA.value)
        try:
            hca_args = [provider] + nodes
            NetworkV().validate('hca', hca_args)
        except Exception as exc:
            check_res.set_fail(checked_target=cfg.ALL_MINIONS,
                               comment=str(exc))
        else:
            check_res.set_passed(checked_target=cfg.ALL_MINIONS,
                                 comment="HCA Presence and "
                                         f"Ports Validated: {hca_args}")
        res.append(check_res)

    return res
```

Reply:
Comment addressed.